src/interfaces/token.ts

```diff
@@ -384,6 +385,10 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
     }
   }
 
+  async findMetadataOverrides(address: Address): Promise<VaultMetadataOverrides> {
+    return this.yearn.services.meta.findOne(address);
+  }
```

Comment:
Lets move it to the Vault Interface instead of Token, since its more according. Also, this data should be available by calling `vault.getDynamic()`... it should be currently overriding metadata with yearn-meta info, please check if that is the behavior.

Reply:
Currently we are getting the vaults data from this endpoint https://api.yearn.finance/v1/chains/1/vaults/all which doesn't seem to have any overrides.

This is what the `yvBOOST` json looks like:

```json
{
  "inception": 12194927,
  "address": "0x9d409a0A012CFbA9B15F6D4B36Ac57A46966Ab9a",
  "symbol": "yvBOOST",
  "name": "yvBOOST 0.3.5",
  "display_name": "yveCRV",
  "icon": "https://rawcdn.githack.com/yearn/yearn-assets/3bfe6d4c0422f86046ce7e69ed459186bbde7da6/icons/multichain-tokens/1/0x9d409a0A012CFbA9B15F6D4B36Ac57A46966Ab9a/logo-128.png",
  "token": {
    "name": "veCRV-DAO yVault",
    "symbol": "yveCRV-DAO",
    "address": "0xc5bDdf9843308380375a611c18B50Fb9341f502A",
    "decimals": 18,
    "display_name": "yveCRV",
    "icon": "https://rawcdn.githack.com/yearn/yearn-assets/3bfe6d4c0422f86046ce7e69ed459186bbde7da6/icons/multichain-tokens/1/0xc5bDdf9843308380375a611c18B50Fb9341f502A/logo-128.png"
  },
  "tvl": {
    "total_assets": 1.3007427267952816e25,
    "price": 1.747124020003843,
    "tvl": 22725588.618293326
  },
  "apy": {
    "type": "v2:averaged",
    "gross_apr": 0.8358165527057161,
    "net_apy": 1.1103332454716277,
    "fees": {
      "performance": 0.1,
      "withdrawal": null,
      "management": 0,
      "keep_crv": null,
      "cvx_keep_crv": null
    },
    "points": {
      "week_ago": 2.3658360810776955,
      "month_ago": 1.1103332454716277,
      "inception": 0.5297655231251615
    },
    "composite": null
  },
  "strategies": [
    {
      "address": "0xd7240B32d24B814fE52946cD44d94a2e3532E63d",
      "name": "StrategyYearnVECRV"
    }
  ],
  "endorsed": true,
  "version": "0.3.5",
  "decimals": 18,
  "type": "v2",
  "emergency_shutdown": false,
  "updated": 1648364507,
  "migration": {
    "available": false,
    "address": "0x9d409a0A012CFbA9B15F6D4B36Ac57A46966Ab9a"
  }
}

```

Hence why I've added this new function in the sdk, so we can fetch the overrides for a particular address. I could have a knowledge gap here, but I don't see where we are getting the yearn-meta info for the labs tokens. Do you know @xgambitox?

Reply:
We are currently not fetching any data for labs, and that is part of the problem, since all is done in FE. (should be ported later)

What I would expect from `vault.getDynamic()` method, is to get all vault data defined as dynamic (data that can change in a in a period of time) with corresponding overrides already merged. If this is not the current case, we should aim to refactor it and fix it. We expect both `vault.getDynamic()` and `vault.get()` include overrides merged.

Reply:
Sounds good, removed with https://github.com/yearn/yearn-sdk/pull/263/commits/69fa49237a012eb0028d8fb6e5f968853aef1670 and refactored the frontend to use the `vaults.getDynamic()` with https://github.com/yearn/yearn-finance-v3/pull/558